storage-manager/src/Replicator.cpp

```diff
@@ -238,26 +299,32 @@ int Replicator::addJournalEntry(const boost::filesystem::path &filename, const u
         {
             bHeaderChanged = true;
             string header = (boost::format("{ \"version\" : \"%03i\", \"max_offset\" : \"%011u\" }") % version % thisEntryMaxOffset).str();
-            err = ::pwrite(fd, header.c_str(), header.length() + 1,0);
+            err = _pwrite(fd, header.c_str(), header.length() + 1,0);
+            l_errno = errno;
             repHeaderDataWritten += (header.length() + 1);
             if ((uint)err != (header.length() + 1))
             {
                 // only the header was possibly changed rollback attempt
                 mpLogger->log(LOG_CRIT, "Replicator::addJournalEntry: Updating journal header failed. "
                     "Attempting to rollback and continue.");
-                int rollbackErr = ::pwrite(fd, headerRollback.c_str(), headerRollback.length() + 1,0);
+                int rollbackErr = _pwrite(fd, headerRollback.c_str(), headerRollback.length() + 1,0);
                 if ((uint)rollbackErr == (headerRollback.length() + 1))
                     mpLogger->log(LOG_CRIT, "Replicator::addJournalEntry: Rollback of journal header success.");
                 else
                     mpLogger->log(LOG_CRIT, "Replicator::addJournalEntry: Rollback of journal header failed!");
-                return err;
+                errno = l_errno;
+                if (err < 0)
+                    return err;
+                else
+                    return 0;
```

Comment:
Since no user data has been written yet, it should either return 0 or -1 indicating an error.  Returning a val > 0 means to the caller that some user data was written successfully.